crates/egui/src/widgets/slider.rs

```diff
@@ -616,18 +626,37 @@ impl<'a> Slider<'a> {
             let rail_radius = ui.painter().round_to_pixel(self.rail_radius_limit(rect));
             let rail_rect = self.rail_rect(rect, rail_radius);
 
-            let position_1d = self.position_from_value(value, position_range);
-
             let visuals = ui.style().interact(response);
+            let widget_visuals = &ui.visuals().widgets;
+
             ui.painter().add(epaint::RectShape {
                 rect: rail_rect,
-                rounding: ui.visuals().widgets.inactive.rounding,
-                fill: ui.visuals().widgets.inactive.bg_fill,
+                rounding: widget_visuals.inactive.rounding,
+                fill: widget_visuals.inactive.bg_fill,
                 stroke: Default::default(),
             });
 
+            let position_1d = self.position_from_value(value, position_range);
             let center = self.marker_center(position_1d, &rail_rect);
 
+            // Paint trailing color if enabled.
+            if self.trailing_color {
+                let mut trailing_rail_rect = rail_rect;
+
+                // The trailing rect has to be drawn differently depending on the orientation.
+                match self.orientation {
+                    SliderOrientation::Vertical => trailing_rail_rect.min.y = center.y,
+                    SliderOrientation::Horizontal => trailing_rail_rect.max.x = center.x,
+                };
+
+                ui.painter().add(epaint::RectShape {
+                    rect: trailing_rail_rect,
+                    rounding: widget_visuals.inactive.rounding,
+                    fill: ui.visuals().selection.bg_fill,
+                    stroke: Default::default(),
+                });
```

Comment:
style: this could be slightly shorter 🤷‍♂️

```suggestion
                ui.painter().rect_filled(
                    trailing_rail_rect,
                    widget_visuals.inactive.rounding,
                    ui.visuals().selection.bg_fill,
                ));
```